src/protocol.h

```diff
@@ -145,8 +145,32 @@ class CInv
     {
         int nVersion = s.GetVersion();
         READWRITE(type);
+
+        // The implicit P2P network protocol inherited from Bitcoin Core has
+        // zcashd nodes sort-of ignoring unknown CInv message types in inv
+        // messages: they are added to the known transaction inventory, but
+        // AlreadyHave returns true, so we do nothing with them. Meanwhile for
+        // getdata messages, ProcessGetData ignores unknown message types
+        // entirely.
+        switch (type) {
+        case MSG_TX:
+        case MSG_BLOCK:
+        case MSG_FILTERED_BLOCK:
+            break;
+        case MSG_WTX:
+            if (nVersion < CINV_WTX_VERSION) {
+                throw std::ios_base::failure(
+                    "Negotiated protocol version does not support CInv message type MSG_WTX");
+            }
+            break;
+        default:
+            // This includes UNDEFINED, which should never be serialized.
+            throw std::ios_base::failure("Unknown CInv message type");
```

Comment:
These `throw`s result in `zcashd` printing to `stderr`, which the RPC test framework interprets as a test failure. How should we handle this?

- `ProcessMessages` has a few exception catches for "allowed" IO failures (namely under-length or over-length messages). We could add one to stop it printing the error message in this case?
- We could alter the RPC test framework to allow annotating tests with a `pass_with_stderr` boolean?

Reply:
In what test do we need to treat this as not-a-failure?

Reply:
In a pairing, we worked out that it should be possible to capture the `zcashd` nodes' `stderr` from within the RPC test, allowing the test itself to pass (instead of getting caught by the test handler), so we'd just do that.